tests/test_instance_id.py

```
@@ -0,0 +1,160 @@
+
+# ---------------------------------------------------------------------------
+# Backward compatibility: existing config without instance_id
+# ---------------------------------------------------------------------------
+
+
+class TestBackwardCompatibility:
+    """Ensure old configs without instance_id load and validate fine."""
+
+    def test_old_config_without_instance_id(self):
+        old_config = {
+            "host": "0.0.0.0",
```

Ruff (0.15.10)

[error] Possible binding to all interfaces

(S104)